### PR TITLE
Etcd refactor

### PR DESCRIPTION
**Pull request by stephenrlouie**
The refactor breaks down the etcd role into smaller component plays that could be reused in larger playbooks, such as cluster wide, deployments, upgrades, etc. If all the roles are more modular it would make additional playbooks much easier to create and this gives us a consistent pattern to follow with the other roles. 

Hit an ansible bug: https://github.com/ansible/ansible/issues/13818 that effects ansible > 2.0, so I rolled back to 1.9.4, and this works. The bug makes ansible unable to find the templates/files in the role when run from Vagrant.

- Related to #793
- Should make #852 easier
- An example of what #838 should be

----------

Would appreciate feedback @danehans @rutsky @adamschaub 

Refactor of etcd role to be modeled after [openstack-ansible](https://github.com/openstack/openstack-ansible/tree/kilo) and [hpe-helion] (https://github.com/hpe-helion-os/helion-ansible). Confirmed on Virtualbox + CoreOS and Virtualbox + Centos

- I think we should always use the binaries from github, regardless of OS, so we can always control the version of etcd present. CoreOS comes with etcd, we would have to uninstall and install the specified version for CoreOS. Where Centos uses the [generic package installer] (http://docs.ansible.com/ansible/package_module.html) as seen [here] (https://github.com/kubernetes/contrib/blob/master/ansible/roles/etcd/tasks/main.yml#L2). It does not support versions, we currently just specify `latest` and end up with the latest copy of etcd that is in yum/apt-get. (A very unlucky person could be running the install script and the version served by yum/apt-get might get changed during their deployment, leaving them with different versions of etcd) Ubuntu is the only one using the binaries right now, seen [here] (https://github.com/kubernetes/contrib/blob/master/ansible/roles/etcd/tasks/main.yml#L11). Using binaries will let anyone regardless of OS to use a specific version of etcd.

Design Question:
- Not sure where `deploy.yml` should live. I put it one level up from `tasks` right now since its wraps a bunch of `etcd/tasks` playbooks together
- How granular do we want bottom level playbooks that live in the task file.
  - `install.yml` has all install instructions regardless of OS.
  - `install.yml` just cycles through each OS supported, and calls plays like `install-coreos.yml` `install-ubuntu.yml`
    - Right now I'm breaking it out if there is more than 2 statements that apply to the same `when`, see `coreos-start.yml` and `coreos-stop.yml`. I see this is being a bit more manageable down the road at the cost of the directory getting a little crowded.

### CODE REVIEW

**Comment by stephenrlouie** [comment]
@danehans @adamschaub - could you take a look, no longer a WIP

**Comment by stephenrlouie** [comment]
@eparis & @danehans 

Could you guys provide feedback for this ansible refactor. I think more granular tasks will make our lives easier down the road. I modeled this after hpe-helion & openstack-ansible. Adding functionality for upgrades / uninstall and whatever comes in the future should be more manageable like this as opposed to what was there before.

 
I confirmed this works on Virtualbox + CoreOS and Virtualbox + Centos. I'll be submitting another PR soon to support updates for CoreOS

File: ansible/vagrant/Vagrantfile [hunk 1]
```diff
@@ -159,7 +159,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
     set_common_ansible_options(ansible)
 
     # This sets up both flannel and kube.
-    ansible.playbook = "../cluster.yml"
+    ansible.playbook = "../deploy.yml"
```
Comment (adamschaub) [inline ansible/vagrant/Vagrantfile]: Should this be ../playbooks/deploy.yml?

Comment (stephenrlouie) [inline ansible/vagrant/Vagrantfile]: good catch

**Comment by stephenrlouie** [comment]
The second commit organizes the top level playbooks into a `playbooks` directory. This causes vagrant to fail because ansible cannot find `group_vars`. 

The fix is to move `group_vars` into the directory where the playbook is being called so to `contrib/ansible/playbooks` **or** where the inventory file lives for vagrant `contrib/ansible/vagrant/.vagrant/provisioners/ansible/inventory/vagrant_ansible_inventory`

A different PR: https://github.com/kubernetes/contrib/pull/933 is related and puts a symlink to `group_vars` where the `vagrant_ansible_inventory` file lives.

**Pull request closed without merging**